base_user_role/views/user.xml

```diff
@@ -12,7 +12,7 @@
             <page string="Roles">
                 <field name="role_line_ids" nolabel="1">
                     <tree editable="bottom" colors="grey: not is_enabled;">
-                        <field name="role_id"/>
+                        <field name="role_id" required="1"/>
```

Comment:
I agree, `role_id` should be required in `res.users.role.line` (and `user_id` too I think, with `ondelete='cascade'` as well).